content/blog/rn-bridging-native-modules/index.md

```diff
@@ -0,0 +1,341 @@
+
+
+```swift
+// MyWebView.swift
+
+...
+
+@objc(MyWebView)
+public class MyWebView: UIView, WKNavigationDelegate, WKUIDelegate {
+    var webView: WKWebView!
+    var webViewURL: NSString = "https://beanloop.se"
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        let configuration: WKWebViewConfiguration = WKWebViewConfiguration()
+        webView = WKWebView(frame: frame, configuration: configuration)
+        if let url = URL(string: self.webViewURL as String) {
+            webView.load(URLRequest(url: url))
+            webView.allowsBackForwardNavigationGestures = true
+        }
+
+        addSubview(webView)
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+```
+
+## Bridge to React Native with Objective-C
+
+To bridge the WebView to React Native we need to use Objective-C as an intermediary. This is where the `@objc` attributes will come in handy.   
+
+- Directly in your iOS project create a new Objective-C file.
+    - Right-click your project folder and select **New File...**
+    - Select **Objective-C File**, then **Next**.
+    - Name it the same as the Swift file, in our case **MyWebView**. Make sure the file will be saved in the **ios** folder. Also, make sure the project folder is selected in **Group**. Select your project target.
+
+*Note: In Objective-C, files with the extension `.h` are header files, used for public declarations of the class. Files with the extension `.m` are implementation files, which contain the private implementation of the class.*
+
+```m
+// MyWebView.m
+#import <Foundation/Foundation.h>
+#import <React/RCTBridgeModule.h>
+#import <React/RCTViewManager.h>
+
+@interface RCT_EXTERN_MODULE(MyWebViewManager, RCTViewManager)
+@end
+```
+
+That is all we need to bridge our very simple native WebView to React Native. Now, let's switch to our React Native code to try it out!
+
+## React Native
+
+```js
+import React from "react"
+import { View, requireNativeComponent, NativeModules } from "react-native"
+import { WebView } from "react-native-webview"
+
+const MyWebView = requireNativeComponent("MyWebView")
+const { MyWebViewManager } = NativeModules
+
+const MyRNWebView = () => {
+    return (
+        <View style={{ flex: 1 }}>
+            <WebView
+                nativeConfig={{
+                    component: MyWebView,
+                    viewManager: MyWebViewManager,
+                }}
+            />
+        </View>
+    )
+}
+```
+
+Here we use the regular `WebView` component from `react-native-webview` but we pass our own native components as the `component` & `viewManager` in the `nativeConfig`, as per [React Native's docs](https://facebook.github.io/react-native/docs/webview#nativeconfig).
+
+![WebView in action example](./rn-webview-example.png)
```

Comment:
correct the reference path to the image

Reply:
This is apparently correct. Must just be GitHub that can't view the image properly.